scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java

```diff
@@ -128,12 +140,76 @@ public SchedulingService(SchedulingInfrastructure infrastructure, SchedulerState
 
         pinger = new NodePingThread(this);
         pinger.start();
+
+        if (PASchedulerProperties.SCHEDULER_AUTOMATIC_REMOVED_JOB_DELAY.getValueAsInt() > 0) {
+            startHouseKeeping(this.jobsToDeleteFromDB, this.infrastructure);
+        }
+    }
+
+    public void startHouseKeeping(final ConcurrentLinkedQueue<JobId> jobsQueue,
+            final SchedulingInfrastructure infrastructure) {
+        houseKeepingScheduler = new Scheduler();
+        String cronExpr = "* * * * *";
+        if (PASchedulerProperties.SCHEDULER_AUTOMATIC_REMOVED_JOB_CRON_EXPR.isSet()) {
+            cronExpr = PASchedulerProperties.SCHEDULER_AUTOMATIC_REMOVED_JOB_CRON_EXPR.getValueAsString();
+        }
+        houseKeepingScheduler.schedule(cronExpr, new Runnable() {
+            @Override
+            public void run() {
+                long timeNow = System.currentTimeMillis();
+                List<JobId> jobIdList = infrastructure.getDBManager().getJobsToRemove(timeNow);
+                List<Long> longJobIdList = new ArrayList<Long>(jobIdList.size());
+
+                // remove from the memory context
+                long inMemoryTimeStart = System.currentTimeMillis();
+                for (JobId jobId : jobIdList) {
+                    TerminationData terminationData = jobs.removeJob(jobId);
+                    submitTerminationDataHandler(terminationData);
+                    InternalJob job = getInfrastructure().getDBManager().loadJobWithTasksIfNotRemoved(jobId);
+                    if (job != null) {
+                        job.setRemovedTime(System.currentTimeMillis());
+                        ServerJobAndTaskLogs.remove(jobId);
+                        getListener().jobStateUpdated(job.getOwner(),
+                                                      new NotificationData<JobInfo>(SchedulerEvent.JOB_REMOVE_FINISHED,
+                                                                                    new JobInfoImpl((JobInfoImpl) job.getJobInfo())));
+                        wakeUpSchedulingThread();
+                    }
+                    longJobIdList.add(jobId.longValue());
+                }
+                long inMemoryTimeStop = System.currentTimeMillis();
+
+                // set the removedTime and also remove if required by the JOB_REMOVE_FROM_DB setting
+                long dbTimeStart = System.currentTimeMillis();
+                if (!longJobIdList.isEmpty()) {
+                    infrastructure.getDBManager()
+                                  .executeHousekeepingInDB(longJobIdList,
+                                                           PASchedulerProperties.JOB_REMOVE_FROM_DB.getValueAsBoolean());
+                }
+                long dbTimeStop = System.currentTimeMillis();
+                logger.debug("HOUSEKEEPING " + timeNow + " finished ! " + "(Hibernate context removal took " +
```

Comment:
I would see this message at INFO level : "Housekeeping of jobs job1 job2 job3 ... performed, (Hibernate context removal took ... )